src/main/java/am/ik/rsocket/InteractionModel.java

```diff
@@ -70,6 +86,19 @@ Publisher<?> request(RSocket rsocket, Args args) {
 		}
 	};
 
+	private static Consumer<Payload> writeToOutput(Args args) {
+		return (payload) -> args.outputFile().map(path -> {
+			ByteBuf buf = payload.data();
+			byte[] bytes = new byte[buf.readableBytes()];
+			buf.getBytes(buf.readerIndex(), bytes);
+			try {
+				return Files.write(path, bytes);
```

Comment:
Using `Files.write` for each payload looks quite inefficient

Reply:
> Maybe a rewrite would be your preference here?

If I work on this issue, I would take that approach.

Reply:
Understood.